modules/apps/portal-workflow/portal-workflow-kaleo-service/src/main/java/com/liferay/portal/workflow/kaleo/service/persistence/impl/KaleoTaskInstanceTokenFinderImpl.java

```diff
@@ -448,13 +447,10 @@ protected String getAssigneeClassPKs(
 		sb.append("AND (KaleoTaskAssignmentInstance.assigneeClassPK IN (");
 
 		sb.append(
-			Stream.of(
-				assigneeClassPKs
-			).map(
-				String::valueOf
-			).collect(
-				Collectors.joining(StringPool.COMMA_AND_SPACE)
-			));
+			StringUtil.merge(
```

Comment:
@JanisZhang The `StringBundler` usage is wrong before/after the change. The `StringBundler` is initialized with size of `array.length * 2 + 1`, but three's only 3 appends before calling `sb.toString()`.

For performance optimization, you should keep the initial size, use a loop to populate every class pk string form, and COMMA_AND_SPACE, reset the index to index - 1, and append the last element.